### PR TITLE
fix(hotreload): bound reload request body size

### PR DESCRIPTION
**Pull request by Aias00**
## What

Limit the HTTP hot reload request body to 1 MiB before parsing reload YAML, and return `413 Request Entity Too Large` when the limit is exceeded. The reload token comparison now uses constant-time comparison.

## Why

The handler previously used `io.ReadAll(r.Body)` without a size limit, which allowed an authenticated caller to force excessive memory use with a large reload body.

## Testing

- `go test ./pkg/hotreload`
- `git diff --check`

### CODE REVIEW

File: pkg/hotreload/http_handler.go [hunk 1]
```diff
@@ -95,14 +98,20 @@ func (h *ReloadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 			logger.Errorf("Failed to read request body: %v", readErr)
 			http.Error(w, fmt.Sprintf("Failed to read request body: %v", readErr), http.StatusBadRequest)
 			return
 		}
 
+		if len(content) > maxReloadBodyBytes {
+			logger.Warnf("Reload request body from %s exceeded %d bytes", r.RemoteAddr, maxReloadBodyBytes)
+			http.Error(w, "Request body too large", http.StatusRequestEntityTooLarge)
+			return
+		}
```
[the remark this answers is not in the document]
Comment (Aias00) [reply pkg/hotreload/http_handler.go]: Addressed in a3381743 by adding tests for the exact max-size boundary and empty-body fallback behavior.

File: pkg/hotreload/http_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+import (
+	"github.com/stretchr/testify/assert"
+)
```
[the remark this answers is not in the document]
Comment (Aias00) [reply pkg/hotreload/http_handler_test.go:29]: Thanks. This repository enforces imports-formatter in CI rather than plain gofmt import grouping. I kept the current grouping because it matches the formatter used by the repository checks.

File: pkg/hotreload/http_handler.go [hunk 3]
```diff
@@ -95,14 +98,20 @@ func (h *ReloadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Try to read from body first (handles chunked encoding where ContentLength == -1)
 	// If body is empty, fallback to file reload
 	if r.Body != nil {
-		content, readErr := io.ReadAll(r.Body)
+		content, readErr := io.ReadAll(io.LimitReader(r.Body, maxReloadBodyBytes+1))
```
[the remark this answers is not in the document]
Comment (Aias00) [reply pkg/hotreload/http_handler.go]: Addressed in a3381743 by replacing LimitReader with http.MaxBytesReader and handling http.MaxBytesError as 413.

**Comment by Aias00** [comment]
Thanks for the Codecov report. Coverage is passing and all modified coverable lines are covered.

**Comment by Aias00** [comment]
close : https://github.com/apache/dubbo-go-pixiu/issues/975

**Review by AlexStocks — APPROVED** [review]
